parsers/cxx/cxx_parser_tokenizer.c

```diff
@@ -1359,7 +1367,11 @@ bool cxxParserParseNextToken(void)
 						// Exclude possible cases of recursive macro expansion that
 						// causes a single token chain to grow too big
 						//    -D'x=y.x'
-						(iInitialTokenChainSize < CXX_PARSER_MAXIMUM_TOKEN_CHAIN_SIZE)
+						(iInitialTokenChainSize < CXX_PARSER_MAXIMUM_TOKEN_CHAIN_SIZE) &&
+						// Detect other cases of nasty macro expansion that cause
+						// the unget buffer to grow fast (but the token chain to grow slowly)
+						//    -D'p=a' -D'a=p+p'
+						(cppUngetBufferSize() < CXX_PARSER_MAXIMUM_UNGET_BUFFER_SIZE_FOR_MACRO_REPLACEMENTS)
```

Comment:
This hack could probably be removed then. It doesn't hurt much... but well, as you wish.

Reply:
If we use 1024 as the definition for CXX_PARSER_MAXIMUM_MACRO_USE_COUNT, your hack is still an important stopper of expansion.

Reply:
Though I didn't change CXX_PARSER_MAXIMUM_MACRO_USE_COUNT, I kept your code as a safe-stopper.